src/Akka.Persistence.Sql/Snapshot/ByteArraySnapshotDao.cs

```diff
@@ -48,6 +107,181 @@ public class ByteArraySnapshotDao : ISnapshotDao
             _readIsolationLevel = snapshotConfig.ReadIsolationLevel;
 
             _shutdownCts = new CancellationTokenSource();
+            _pendingLatestChannel = Channel.CreateUnbounded<LatestSnapRequestEntry>();
+            int maxSubStreamsForReads = 8; // TODO: Configurable
+            int maxRequestsPerBatch = 50;
+            _latestSnapStream = Source.ChannelReader(_pendingLatestChannel.Reader)
+                .GroupBy(maxSubStreamsForReads, a=> a.PersistenceId.GetHashCode()% maxSubStreamsForReads)
+                .BatchWeighted(
+                    maxRequestsPerBatch,
+                    a => 1,
+                    e =>
+                    {
+                        var a = new QueryLatestSnapSet();
+                        a.Add(e);
+                        return a;
+                    },
+                    (a, e) =>
+                    {
+                        a.Add(e);
+                        return a;
+                    })
+                .SelectAsync(1,
+                    async a =>
+                    {
+
+                        using (var connection = _connectionFactory.GetConnection())
+                        {
+                            if (connection.UseDateTime)
+                            {
+                                //TODO: Consolidate/fixup different rowtype issues.
```

Comment:
I am admittedly unsure how we got to having separate categories here; we -should- be able to set all of this up such that either L2DB is doing the 'right thing' for a given db config, and/or we sanely do our queries against whatever mapping is there. (well, actually it's a bit of both... still...)